testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/concurrent/HungTasksTerminationTestCase.java

```diff
@@ -0,0 +1,301 @@
+
+    private static final PathAddress EE_SUBSYSTEM_PATH_ADDRESS = PathAddress.pathAddress(PathElement.pathElement(SUBSYSTEM, EeExtension.SUBSYSTEM_NAME));
+    private static final String RESOURCE_NAME = HungTasksTerminationTestCase.class.getSimpleName();
+    // a task is considered hung when is running over 0.5s
+    private static final long HUNG_TASK_THRESHOLD_TEST_VALUE = 500;
+    // hung tasks periodic cancellation should be done every 5s
+    private static final long HUNG_TASK_CANCELLATION_PERIOD_TEST_VALUE = 5000;
+
+    @ArquillianResource
+    private ManagementClient managementClient;
+
+    @Deployment
+    public static Archive<?> deploy() {
+        return ShrinkWrap.create(JavaArchive.class, HungTasksTerminationTestCase.class.getSimpleName() + ".jar")
+                .addClasses(HungTasksTerminationTestCase.class)
+                .addAsManifestResource(new StringAsset("Dependencies: org.jboss.as.controller, org.jboss.as.ee, org.jboss.remoting\n"), "MANIFEST.MF")
+                .addAsManifestResource(createPermissionsXmlAsset(
+                        new RemotingPermission("createEndpoint"),
+                        new RemotingPermission("connect"),
+                        new FilePermission(System.getProperty("jboss.inst") + "/standalone/tmp/auth/*", "read")
+                ), "permissions.xml");
+    }
+
+    @Test
+    public void testManagedExecutorServiceHungTasksCancellationOperation() throws Exception {
+        final PathAddress pathAddress = EE_SUBSYSTEM_PATH_ADDRESS.append(EESubsystemModel.MANAGED_EXECUTOR_SERVICE, RESOURCE_NAME);
+        // add
+        final ModelNode addOperation = Util.createAddOperation(pathAddress);
+        final String jndiName = "java:jboss/ee/concurrency/executor/" + RESOURCE_NAME;
+        addOperation.get(ManagedExecutorServiceResourceDefinition.JNDI_NAME).set(jndiName);
+        addOperation.get(ManagedExecutorServiceResourceDefinition.HUNG_TASK_THRESHOLD).set(HUNG_TASK_THRESHOLD_TEST_VALUE);
+        final ModelNode addResult = managementClient.getControllerClient().execute(addOperation);
+        Assert.assertFalse(addResult.get(FAILURE_DESCRIPTION).toString(), addResult.get(FAILURE_DESCRIPTION).isDefined());
+        try {
+            final ManagedExecutorService executorService = InitialContext.doLookup(jndiName);
+            Assert.assertNotNull(executorService);
+            testHungTasksCancellationOperation(pathAddress, executorService);
+        } finally {
+            // remove
+            final ModelNode removeOperation = Util.createRemoveOperation(pathAddress);
+            removeOperation.get(OPERATION_HEADERS, ALLOW_RESOURCE_SERVICE_RESTART).set(true);
+            final ModelNode removeResult = managementClient.getControllerClient().execute(removeOperation);
+            Assert.assertFalse(removeResult.get(FAILURE_DESCRIPTION).toString(), removeResult.get(FAILURE_DESCRIPTION)
+                    .isDefined());
+        }
+    }
+
```

Comment:
@emmartins what about a short description:

/**
     * Tests the `terminate-hung-tasks` operation works on a `managed-scheduled-executor-service`
     * @throws Exception
     */

Reply:
Used a different wording, to not depend on attributes/operation names, but done.